src/main/java/cn/nukkit/blockentity/BlockEntityChest.java

```diff
@@ -55,7 +55,6 @@ protected void initBlockEntity() {
     @Override
     public void close() {
         if (!closed) {
-            unpair();
```

Comment:
I'm not sure but maybe `doubleInventory` of both chests should be still set `null` here?

Reply:
Yeah I think you're right but I also think the unpair was added there to stop chests freezing open which I've seen occasionally tonight investigating your comments. At least unparing means they get "fixed" when the chunk unloads/reloads. My "fix" just masked the root cause of the chunk spanning chest bug I guess. Back to the drawing board, thanks for the review :)

Reply:
Hey @PetteriM1,
I think the original fix is good as it is.
I looked into nulling doubleInventory on both chests but this was a problem when unloading both chunks and a chests pair had already been unloaded, i got into NPEs trying to access the already unloaded pair of this chest.
While testing I did see that the doubleInventory references in both chests were being nulled during the chunk unload and I assume the doubleInventory would then be GC'd?
Anyway I tested both the chunk unload and chest break scenarios fully and I'm confident the results are good.
The following are the testcases/steps I ran for each scenario, detailed results/logs are in the attached files. Please let me know if you need any more info :)

[resultTestsetBreakChest.txt](https://github.com/CloudburstMC/Nukkit/files/6490456/resultTestsetBreakChest.txt)
[resultTestsetChunkUnload.txt](https://github.com/CloudburstMC/Nukkit/files/6490457/resultTestsetChunkUnload.txt)

Scenario: Chunk unload (close)
resultTestsetChunkUnload.txt(2): ## TESTCASE - Both Chunks Load on login ##
resultTestsetChunkUnload.txt(8): ## Both chunks load on login ##
resultTestsetChunkUnload.txt(12): ## Initial open right chest ##
resultTestsetChunkUnload.txt(29): ## Re-open right chest ##
resultTestsetChunkUnload.txt(43): ## Initial open left chest ##
resultTestsetChunkUnload.txt(60): ## Re-open left chest ##
resultTestsetChunkUnload.txt(75): ## TESTCASE - Both Chunks Unload/Reload ##
resultTestsetChunkUnload.txt(76): ## Move distance enough to unload both chunks ##
resultTestsetChunkUnload.txt(78): ## Left chunk unload ##
resultTestsetChunkUnload.txt(95): ## Right chunk unload ##
resultTestsetChunkUnload.txt(112): ## Return to test area ##
resultTestsetChunkUnload.txt(113): ## Both chunks reload ##
resultTestsetChunkUnload.txt(117): ## Initial open left chest ##
resultTestsetChunkUnload.txt(134): ## Re-open left chest ##
resultTestsetChunkUnload.txt(148): ## Initial open right chest ##
resultTestsetChunkUnload.txt(165): ## Re-open right chest ##
resultTestsetChunkUnload.txt(180): ## TESTCASE - Left Chunk Unload, Right Chunk Remains Loaded ##
resultTestsetChunkUnload.txt(182): ## Move distance enough to unload left chunk only (192.0, 70.0, 16.0)##
resultTestsetChunkUnload.txt(183): ## Left chunk unload ##
resultTestsetChunkUnload.txt(200): ## Return to test area ##
resultTestsetChunkUnload.txt(201): ## Left chunk reloads ##
resultTestsetChunkUnload.txt(204): ## Initial open left chest ##
resultTestsetChunkUnload.txt(221): ## Re-open left chest ##
resultTestsetChunkUnload.txt(235): ## Initial open right chest ##
resultTestsetChunkUnload.txt(250): ## TESTCASE - Both Chunks Unload/Reload2 ##
resultTestsetChunkUnload.txt(252): ## Move distance enough to unload both chunks ##
resultTestsetChunkUnload.txt(253): ## Left chunk unload ##
resultTestsetChunkUnload.txt(270): ## Right chunk unload ##
resultTestsetChunkUnload.txt(288): ## Return to test area ##
resultTestsetChunkUnload.txt(289): ## Both chunks reload ##
resultTestsetChunkUnload.txt(293): ## Initial open right side double chest ##
resultTestsetChunkUnload.txt(310): ## Re-open right side double chest ##
resultTestsetChunkUnload.txt(324): ## Initial open left side double chest
resultTestsetChunkUnload.txt(341): ## Re-open left side double chest
resultTestsetChunkUnload.txt(361): ## Unload left chunk ##
resultTestsetChunkUnload.txt(378): ## Unload right chunk ##

Scenario Break chest pair (onBreak)
resultTestsetBreakChest.txt(7): ## PreTest - ChestState ##
resultTestsetBreakChest.txt(8): ## Initial open right chest ##
resultTestsetBreakChest.txt(25): ## Re-open right chest ##
resultTestsetBreakChest.txt(39): ## Initial open left chest ##
resultTestsetBreakChest.txt(56): ## Re-open right chest ##
resultTestsetBreakChest.txt(71): ## TESTCASE - Unpair on Break (left chest) ##
resultTestsetBreakChest.txt(73): ## Break left chest ##
resultTestsetBreakChest.txt(99): ## Open right chest ##
resultTestsetBreakChest.txt(114): ## TESTCASE - Re-Pair on place (left chest) ##
resultTestsetBreakChest.txt(116): ## Place left chest ##
resultTestsetBreakChest.txt(125): ## Initial open left chest ##
resultTestsetBreakChest.txt(142): ## Re-open left chest ##
resultTestsetBreakChest.txt(156): ## Initial open right chest ##
resultTestsetBreakChest.txt(171): ## TESTCASE - Unpair on Break (left chest) ##
resultTestsetBreakChest.txt(173): ## Break left chest ##
resultTestsetBreakChest.txt(199): ## Open left chest ##
resultTestsetBreakChest.txt(214): ## TESTCASE - Re-Pair on place (right chest) ##
resultTestsetBreakChest.txt(216): ## Place right chest ##
resultTestsetBreakChest.txt(225): ## Initial open right chest ##
resultTestsetBreakChest.txt(242): ## Re-open right chest ##
resultTestsetBreakChest.txt(256): ## Initial open left chest ##
resultTestsetBreakChest.txt(270): ## chunk unload after log off ##